app/src/test/java/com/codesoom/assignment/application/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,177 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.dto.LoginData;
+import com.codesoom.assignment.errors.UserNotFoundException;
+import com.codesoom.assignment.errors.WrongPasswordException;
+import com.codesoom.assignment.utils.JwtUtil;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.security.SignatureException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+@Nested
+@DisplayName("AuthenticationService의")
+class AuthenticationServiceTest {
+
+    private static final String SECRET = "12345678901234567890123456789010";
+    private static final String AUTH_SCHEME = "Bearer";
+
+    private JwtUtil jwtUtil;
+
+    private AuthenticationService authenticationService;
+
+    private UserService userService;
+
+    @BeforeEach
+    void setUp() {
+        this.jwtUtil = new JwtUtil(SECRET);
+        this.userService = mock(UserService.class);
+        this.authenticationService = new AuthenticationService(jwtUtil, this.userService);
+    }
+
+    @Nested
+    @DisplayName("login 메소드는")
+    class DescribeLogin {
+
+        private User user;
+        private LoginData loginData;
+
+        @BeforeEach
+        void userSetUp() {
+            user = User.builder()
+                    .name("Jack")
+                    .email("jack@email.com")
+                    .password("qwer1234")
+                    .build();
+            loginData = LoginData.builder()
+                    .email(user.getEmail())
+                    .password(user.getPassword())
+                    .build();
+        }
+
+        @Nested
+        @DisplayName("올바른 유저 인증 정보를 받았을 때")
+        class ContextWithValidUserInfo {
+
+            @BeforeEach
+            void setUp() {
+                given(userService.getUser(user.getEmail())).willReturn(user);
+            }
```

Comment:
준비 코드를 보면 올바른 유저 인증 정보에 대한 내용은 없는 것 같아요. 설명을 다음과 같이 수정할 수 있겠죠.

```suggestion
        @DisplayName("찾을 수 있는 유저 정보가 주어지면")
        class ContextWithValidUserInfo {

            @BeforeEach
            void setUp() {
                given(userService.getUser(user.getEmail())).willReturn(user);
            }
```